DariaCagle2(obligatory).cpp

```diff
@@ -0,0 +1,204 @@
+  int sum = kZero;
+  std::cin >> number;
+  while (number != kZero) {
+    int digit = number % kTen;
+    sum += digit;
+    number = (number - (number % kTen)) / kTen;
+    ++i;
+  }
+  float mean = static_cast<float>(sum) / i;
+  std::cout << sum << " " << mean << "\n";
+}
+
+void luckyTicket() {
+  std::cout << "Enter a " << kSix << "-digit number:\n";
+  int number;
+  std::cin >> number;
+  int number1 = (number - (number % kThousand)) / kThousand;
+  int number2 = number % kThousand;
+  int sum1 = kZero;
+  while (number1 != kZero) {
+    int digit = number1 % kTen;
+    sum1 += digit;
+    number1 = (number1 - (number1 % kTen)) / kTen;
+  }
+  int sum2 = kZero;
+  while (number2 != kZero) {
+    int digit = number2 % kTen;
+    sum2 += digit;
+    number2 = (number2 - (number2 % kTen)) / kTen;
+  }
+  if (sum1 == sum2)
+    std::cout << "This is a lucky ticket!\n";
+  else {
+    std::cout << "This is not a lucky ticket!\n";
+  }
+}
+
+void reverseNumbers() {
+  int64_t number;
+  std::cout << "Enter any 32x signed number:\n";
+  std::cin >> number;
+  int digit;
+  int64_t reverseNumber = kZero;
+  if (number == abs(number)) {
+    while (number != kZero) {
+      digit = number % kTen;
+      reverseNumber = reverseNumber * kTen + digit;
+      number = (number - (number % kTen)) / kTen;
+    }
+    std::cout << reverseNumber << "\n";
+  } else {
+    number = abs(number);
+    while (number != kZero) {
+      digit = number % kTen;
+      reverseNumber = reverseNumber * kTen + digit;
+      number = (number - (number % kTen)) / kTen;
+    }
+    std::cout << -reverseNumber << "\n";
+  }
+}
+
+/*void sumOfOdd() {
+  constexpr int kOne = 1;
+  constexpr int kFifty = 50;
+  int numberOfElements;
+  do {
+    std::cout << "Enter number of elements from " << kOne << " to " << kFifty
+              << ":\n";
+    std::cin >> numberOfElements;
+  } while ((numberOfElements < kOne) || (numberOfElements > kFifty));
+  int *elements = new int[numberOfElements];
+  std::cout << "Enter " << numberOfElements << " elements from " << -kSixty
+            << " to " << kNinety << ":\n";
+  size_t i = kZero;
+  while (i != numberOfElements) {
+    std::cin >> elements[i];
+    while ((elements[i] < -kSixty) || (elements[i] > kNinety)) {
+      std::cout << "Wrong range! Enter again:\n";
+      std::cin >> elements[i];
+    }
+    i++;
+  }
+  int sum = kZero;
+  for (int count = kZero; count < numberOfElements; ++count) {
+    if (elements[count] % kTwo == kOne) {
+      sum = sum + elements[count];
+    }
+  }
+  std::cout << sum << "\n";
+}*/
+
+void sumOfOdd() {
+  int numberOfElements;
+  do {
+    std::cout << "Enter number of elements from " << kOne << " to " << kFifty
+              << ":\n";
+    std::cin >> numberOfElements;
+  } while ((numberOfElements < kOne) || (numberOfElements > kFifty));
+  std::cout << "Enter " << numberOfElements << " elements from " << -kSixty
+            << " to " << kNinety << ":\n";
+  int sum = 0;
```

Comment:
опять же приучите себя писать 
```suggestion
  int sum{0};
```
если вы будете в привычке инициализировать переменные как вы делаете это сейчас, то в будущем у вас могут быть очень серьезные проблемы

Reply:
я поняла и в следующих заданиях уже пишу с {}